### PR TITLE
add massiveactions

### PR DESCRIPTION
**Pull request by orthagh**
http://glpi-developer-documentation.readthedocs.io/en/massiveactions/devapi/massiveactions.html
http://glpi-developer-documentation.readthedocs.io/en/massiveactions/plugins/massiveactions.html

### CODE REVIEW

File: source/devapi/massiveactions.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Massive Actions
+---------------
+
+.. image:: images/massiveactions.png
+   :alt: Massive action controls
+
+Goals
+^^^^^
+
+Add to itemtypes `search lists <http://glpi-developer-documentation.readthedocs.io/en/master/devapi/search.html>`_:
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: The link destination does not exists yet. Also, it add something to search results, not to serach itsefl, right?

File: source/devapi/massiveactions.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+- a checkbox before each item,
+- a checkbox for select all items checkboxes,
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: s/for/to/

File: source/devapi/massiveactions.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- an `Actions` button to apply modifications to each selected items.
+
+Update item's fields
+^^^^^^^^^^^^^^^^^^^^
+
+The first option of the ``Actions`` button is ``Update``.
+It permits to modify the fields content of the selected items.
+
+The list of fields displayed in the sub list depends on the searchoptions_ of the current itemtype.
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: Bad link, x2

File: source/devapi/massiveactions.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+By default, all searchoptions_ are automatically displayed in this list.
+To forbid this display for one field, you must define the key `massiveaction`` to false in the searchoptions_ declaration, ex:
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: Double backquote opened and not closed here.

Comment (trasher) [inline source/devapi/massiveactions.rst]: Also, avoid shortcuts like "ex" :p

File: source/devapi/massiveactions.rst [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+   $tab[] = [
+      'id'            => '1',
+      'table'         => self::getTable(),
+      'field'         => 'name',
+      'name'          => __('Name'),
+      'datatype'      => 'itemlink',
+      'massiveaction' => false // <- NO MASSIVE ACTION
+   ];
+
+.. _specific:
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: This label is maybe a bit too large...`` _ma_specfiic`` or ``massiveactions_specific`` would be better.

File: source/devapi/massiveactions.rst [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      $action_label = "My new massive action";
+      $actions[$class.MassiveAction::CLASS_ACTION_SEPARATOR.$action_key] = $action_label;
+
+      return $actions;
+   }
+
+A single declaration is defined by these parts:
+
+- a ``classname``
+- a ``separator``: always ``MassiveAction::CLASS_ACTION_SEPARATOR``
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: `the` not 'a'

File: source/devapi/massiveactions.rst [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+- a ``key``
+- and a ``label``
+
+We can have multiple actions for the same class, and we may target different class from our current object.
+
+.. _specificsubform:
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: Maybe a bit more specific, but should be prefixed as well

File: source/devapi/massiveactions.rst [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+            }
+            return;
+      }
+
+      parent::processMassiveActionsForOneItemtype($ma, $item, $ids);
+   }
+
+Besides an instance of MassiveAction class ``$ma``, we have also an instance of the current ``itemtype`` ``$item and the list of selected id ``$ids``.
+
+In this method, we could use some optional utility functions from the ``MassiveAction $ma`` object supplied in parameter :
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: "MassiveAction <link to apidoc>" ``$ma``

File: source/devapi/massiveactions.rst [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+- ``itemDone``, indicates the result of the current ``$id``, see constants of `MassiveAction class <https://forge.glpi-project.org/apidoc/class-MassiveAction.html#constants>`_. If we miss this call, the current ``$id`` will still be considered as OK.
```
Comment (trasher) [inline source/devapi/massiveactions.rst]: constants of should be part of the link

File: source/plugins/massiveactions.rst [hunk 10]
```diff
@@ -0,0 +1,84 @@
+Massive Actions
+---------------
+
+Plugins can use the core's `massive actions <http://glpi-developer-documentation.readthedocs.io/en/master/devapi/massiveactions.html>`_ for its own itemtypes.
```
Comment (trasher) [inline source/plugins/massiveactions.rst]: Not better to link doc page here?

File: source/plugins/massiveactions.rst [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+               = $action_label;
+
+            break;
+      }
+
+   }
+   return $actions;
+}
+
+Next, in the class defined int the definition, we can use the ``showMassiveActionsSubForm`` and ``processMassiveActionsForOneItemtype`` in the same way as `core documentation <http://glpi-developer-documentation.readthedocs.io/en/master/devapi/massiveactions.html#specificsubform>`_:
```
Comment (trasher) [inline source/plugins/massiveactions.rst]: bad link. use
```
:doc:`my text <../devapi/massiveaction>`
```

File: source/devapi/massiveactions.rst (hunk 1 above)
Comment (orthagh) [inline source/devapi/massiveactions.rst]: For link, i though use the final (master) link should be better than use the search pr one, no ? 
To visualisze, you just have to replacer `master` in url by `search`.

Comment (trasher) [inline source/devapi/massiveactions.rst]: Nope, internal links must be internal (see other comments)

**Pull request merged**